### PR TITLE
[KernelGen][Nvidia] Add _scaled_dot_product_attention_math operator with Triton kernel

### PR DESCRIPTION
**Pull request by chx7514**
## Summary
Adds a Triton kernel for `_scaled_dot_product_attention_math`. Computes scaled dot-product attention using the math-based implementation with explicit softmax and returns both the attention output and full attention weight matrix.

## Testing
- Validated against reference on device via `to_reference(inp, True)`
- Tested on: Nvidia
- All 54 test cases pass

## Performance
Test command: `pytest benchmark/test_scaled_dot_product_attention_math.py --level core`

### scaled_dot_product_attention_math

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|------|--------------------|-------------------|---------|
| float16 | [2, 4, 64, 64] | 0.064320 | 0.012416 | 5.180 |
| float16 | [2, 8, 128, 64] | 0.062848 | 0.012288 | 5.115 |
| float16 | [4, 8, 256, 64] | 0.102368 | 0.024544 | 4.171 |
| float16 | [2, 8, 512, 64] | 0.154400 | 0.041280 | 3.740 |
| float16 | [1, 8, 1024, 64] | 0.259056 | 0.074624 | 3.471 |
| float32 | [2, 4, 64, 64] | 0.031840 | 0.012320 | 2.584 |
| float32 | [2, 8, 128, 64] | 0.041312 | 0.013728 | 3.009 |
| float32 | [4, 8, 256, 64] | 0.080352 | 0.037504 | 2.142 |
| float32 | [2, 8, 512, 64] | 0.129248 | 0.065472 | 1.974 |
| float32 | [1, 8, 1024, 64] | 0.227744 | 0.120576 | 1.889 |
| bfloat16 | [2, 4, 64, 64] | 0.063584 | 0.012672 | 5.018 |
| bfloat16 | [2, 8, 128, 64] | 0.063520 | 0.013056 | 4.865 |
| bfloat16 | [4, 8, 256, 64] | 0.101952 | 0.024512 | 4.159 |
| bfloat16 | [2, 8, 512, 64] | 0.154144 | 0.041184 | 3.743 |
| bfloat16 | [1, 8, 1024, 64] | 0.258976 | 0.073696 | 3.514 |

**Mean speedup: 3.771x**

## Implementation Notes
- Two-pass algorithm: first pass computes softmax statistics and output, second pass materializes full attention weights
- Block size (BLOCK_M=32, BLOCK_N=32) chosen to fit within 166KB shared memory limit on T4/A10 GPUs
- Supports causal masking, attention mask, and custom scale factor

## Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia | PASS (54 cases) | 3.771x | Primary |

## Files Changed
- `src/flag_gems/ops/_scaled_dot_product_attention_math.py`: Triton kernel implementation
- `tests/test_scaled_dot_product_attention_math.py`: Accuracy test
- `benchmark/test_scaled_dot_product_attention_math.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: 5.4)

### CODE REVIEW

File: tests/test_scaled_dot_product_attention_math.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    ref_k = utils.to_reference(k)
+    ref_v = utils.to_reference(v)
+
+    ref_out, ref_weights = sdpa_math(ref_q, ref_k, ref_v, scale=scale)
+
+    with flag_gems.use_gems():
+        res_out, res_weights = sdpa_math(q, k, v, scale=scale)
+
+    utils.gems_assert_close(res_out, ref_out, dtype, atol=3e-2)
+    utils.gems_assert_close(res_weights, ref_weights, dtype, atol=3e-2)
```
Comment (bin913) [inline tests/test_scaled_dot_product_attention_math.py:86]: Could we consolidate these multiple tests into a single test?

Comment (chx7514) [reply tests/test_scaled_dot_product_attention_math.py:86]: Consolidated the four test functions into a single parametrized test (test_scaled_dot_product_attention_math), covering equal/differing q-kv seq lens, both is_causal settings, explicit attn_mask, and custom scale values (54 parametrized cases total).

**Review by bin913 — APPROVED** [review]
